src/OpenSolid/Sphere3d.elm

```diff
@@ -0,0 +1,500 @@
+        , Point3d.fromCoordinates (0, 1, 0)
+        , Point3d.fromCoordinates (0, -1, 0)
+        )
+    --> Nothing
+
+-}
+throughPoints : ( Point3d, Point3d, Point3d, Point3d ) -> Maybe Sphere3d
+throughPoints ( p1, p2, p3, p4 ) =
+    Maybe.andThen
+        {-
+           First three points define a circle.
+           All points M on the normal to this circle through the circle's center are equidistant to p1, p2 and p3.
+           Therefore: find this point M such that it is equidistant to p1, p2, p3 and p4,
+           this will be the center of the sphere.
+        -}
+        (\circle ->
+            let
+                normalAxis =
+                    Circle3d.axis circle
+
+                r =
+                    Circle3d.radius circle
+
+                x =
+                    Point3d.distanceFromAxis normalAxis p4
+
+                y =
+                    Point3d.distanceAlong normalAxis p4
+
+                d =
+                    (r ^ 2 - x ^ 2 - y ^ 2) / (-2 * y)
+            in
+            if y /= 0 then
+                Just <|
+                    with
+                        { centerPoint = Point3d.along normalAxis d
+                        , radius = sqrt (r ^ 2 + d ^ 2)
+                        }
+            else
+                Nothing
+        )
+        (Circle3d.throughPoints ( p1, p2, p3 ))
+
+
+{-| Get the center point of a sphere.
+
+    Sphere3d.centerPoint exampleSphere
+    --> Point3d.fromCoordinates ( 1, 2, 1 )
+
+-}
+centerPoint : Sphere3d -> Point3d
+centerPoint (Internal.Sphere3d properties) =
+    properties.centerPoint
+
+
+{-| Get the radius of a sphere.
+
+    Sphere3d.radius exampleSphere
+    --> 3
+
+-}
+radius : Sphere3d -> Float
+radius (Internal.Sphere3d properties) =
+    properties.radius
+
+
+{-| Get the diameter of a sphere.
+
+    Sphere3d.diameter exampleSphere
+    --> 6
+
+-}
+diameter : Sphere3d -> Float
+diameter sphere =
+    2 * radius sphere
+
+
+{-| Get the circumference of a sphere.
+
+    Sphere3d.circumference exampleSphere
+    -> 18.8496
+
+-}
+circumference : Sphere3d -> Float
+circumference sphere =
+    2 * pi * radius sphere
+
+
+{-| Get the volume of a sphere.
+
+    Sphere3d.diameter exampleSphere
+    --> 6
+
+-}
+volume : Sphere3d -> Float
+volume sphere =
+    4 / 3 * pi * radius sphere ^ 3
+
+
+{-| Scale a sphere around a given point by a given scale.
+
+    Sphere3d.scaleAbout Point3d.origin 3 exampleSphere
+    --> Sphere3d.with
+    -->     { centerPoint =
+    -->         Point3d.fromCoordinates ( 3, 6, 3 )
+    -->     , radius = 9
+    -->     }
+
+-}
+scaleAbout : Point3d -> Float -> Sphere3d -> Sphere3d
+scaleAbout point scale sphere =
+    with
+        { centerPoint = Point3d.scaleAbout point scale (centerPoint sphere)
+        , radius = abs (scale * radius sphere)
+        }
+
+
+{-| Rotate a sphere around a given axis by a given angle (in radians).
+
+    exampleSphere
+        |> Sphere3d.rotateAround Axis3d.y (degrees 90)
+    -> Sphere3d.with
```

Comment:
Can you update this to use the standard multiline `-->` format?

Reply:
This was intentional so that it wil not be verified by `elm-verify-example`. I can try to find an example that leads to round numbers.